common/containers/Tabs/GenerateWallet/components/Mnemonic/Mnemonic.tsx

```diff
@@ -129,18 +125,89 @@ export default class GenerateMnemonic extends React.Component<{}, State> {
     }
   };
 
-  private makeWord = (word: WordTuple) => (
-    <Word
-      key={`${word.word}${word.index}`}
-      index={word.index}
-      word={word.word}
-      value={this.state.confirmValues[word.index] || ''}
-      isReadOnly={!this.state.isConfirming}
-      onChange={this.handleConfirmChange}
-    />
-  );
+  private makeWord = (word: WordTuple) => {
+    const { words, confirmValues, isRevealingNextWord, isConfirming } = this.state;
+    const confirmIndex = words.indexOf(word.word);
+    const nextIndex = confirmValues.length;
+    const isNext = confirmIndex === nextIndex;
+    const isRevealed = isRevealingNextWord && isNext;
+    const hasBeenConfirmed = this.getWordConfirmed(word.word);
+
+    return (
+      <Word
+        key={`${word.word}${word.index}`}
+        index={word.index}
+        confirmIndex={confirmIndex}
+        word={word.word}
+        value={confirmValues[word.index] || ''}
+        showIndex={!isConfirming}
+        isNext={isNext}
+        isBeingRevealed={isRevealed}
+        isConfirming={isConfirming}
+        hasBeenConfirmed={hasBeenConfirmed}
+        onClick={this.handleWordClick}
+      />
+    );
+  };
+
+  private handleWordClick = (_: number, value: string) => {
+    const { confirmValues: previousConfirmValues, words, isConfirming } = this.state;
+    const wordAlreadyConfirmed = previousConfirmValues.includes(value);
+    const activeIndex = previousConfirmValues.length;
+    const isCorrectChoice = words[activeIndex] === value;
+
+    if (isConfirming && !wordAlreadyConfirmed && isCorrectChoice) {
+      const confirmValues = previousConfirmValues.concat(value);
+
+      this.setState({ confirmValues });
+    }
+  };
+
+  private getWordConfirmed = (word: string) => this.state.confirmValues.includes(word);
 
   private skip = () => {
     this.setState({ isConfirmed: true });
   };
+
+  private getShuffledWords = (words: string[]) => {
```

Comment:
would prefer to move this out to `utils`

Reply:
@dternyak 
I can do that, do you anticipate it being used elsewhere? Seems fairly specific to this component. If I move it to `/utils`, should it gets its own file, or go in the `helpers` file (which seems to be mainly for BLL`

Reply:
I don't have a strong opinion on this, and I do agree it's likely this won't be used elsewhere. That said, moving it out to `utils` and then adding a test for it seems valuable to me.